community/kernel/src/test/java/org/neo4j/graphdb/factory/GraphDatabaseSettingsTest.java

```diff
@@ -416,4 +417,31 @@ public void shouldStillSupportCustomNameForHttpsConnector() throws Exception
         assertEquals( new ListenSocketAddress( "127.0.0.1", 9000 ),
                 config.get( config.enabledHttpConnectors().get( 0 ).listen_address ) );
     }
+
+    @Test
+    public void validateRetentionPolicy()
+    {
+        String[] validSet =
+                new String[]{"true", "keep_all", "false", "keep_none", "10 files", "10k files", "10K size", "10m txs",
+                        "10M entries", "10g hours", "10G days"};
+
+        String[] invalidSet = new String[]{"invalid", "all", "10", "10k", "10k a"};
+
+        for ( String valid : validSet )
+        {
+            Config.defaults( keep_logical_logs, valid );
```

Comment:
i think we need to add a bit more assertions here

Reply:
Assert on what? It throws if it's invalid.

Reply:
i'm thinking about asserts that Config.defaults(....).get(..property) equals to expected value

Reply:
It's an identity setting which returns exactly the same as you put in. But it does not hurt to have an assert there :)